Firebase implementations - tracking committed algos across devices

minhle10 opened the pull request:
Created a new firebase.jsx file to handle all things related to reading and writing to FireStore

Wrote code to:
- Initialize users into FireStore if they are not already in the database, (adds the user's already owned CDPs to the database as well).
- Write committed amount and timestamp to the database when users commit their algos
- Retrieve the amount of committed algos from the database to display/use on Algo Governance Page.

Note: Also wrote a function in firebase.jsx to update liquidation. For now, it is not being used until we further discuss how we want to implement the tracking of liquidation (even for people not using the web app)

mccabe-david reviewed (commented):
Once that change is made it looks good to me!

minhle10 commented on src/transactions/cdp.js:
```diff
@@ -724,6 +725,7 @@ function updateCommitment(address, id, commitment) {
   let CDPs = getCDPs();
   CDPs[address][id]["committed"] = commitment;
   localStorage.setItem("CDPs", JSON.stringify(CDPs));
+  updateCommitmentFirestore(address, id, commitment)
```
Done!

minhle10 commented:
I fixed the issue we talked about in the meeting by modifying the useEffect and adding 'commitment' as a state in my latest commits. The useEffect is called when the page first loads up **and** whenever the 'refresh' state is changed. I added this so 'refresh' as a state so that it refreshes the page to pull/display the new amount of committed ALGOs from firebase right after someone commits ALGOs.

ElianaTroper reviewed (approved):
Working w/ new keys. We should add a template keys.js file in the future.
